server/src/main/java/org/opensearch/common/blobstore/SoftDeleteableSupportedBlobContainer.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.blobstore;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * An extension of {@link BlobContainer} that adds support to soft delete files
+ *
+ * @opensearch.internal
+ */
+public interface SoftDeleteableSupportedBlobContainer {
```

Comment:
can we make this to just `SoftDeleteableBlobContainer`? Supported is implicit given implementor needs to implement this interface.

Reply:
removed interface now .